### ISSUE TITLE
400 Bad request with `encrypt` option

### ISSUE THREAD

**Issue by brendt** [amphp/aerys#202]
I've created a self-signed certificate for local development with HTTPS. But I'm always getting a 400  Bad request when visiting the server.

My test setup can be found here: https://github.com/brendt/web-performance-playground

The server shows this:

```
[2017-10-29 08:28:54] debug accept 127.0.0.1:60959 on 127.0.0.1:4433 #143
[2017-10-29 08:28:54] debug crypto negotiated (h2) 127.0.0.1:60959
[2017-10-29 08:28:54] debug close 127.0.0.1:60959 #143
[2017-10-29 08:28:54] debug accept 127.0.0.1:60960 on 127.0.0.1:4433 #145
[2017-10-29 08:28:54] debug crypto negotiated (h2) 127.0.0.1:60960
[2017-10-29 08:28:54] debug GET / HTTP/2.0 @ 127.0.0.1:60960
[2017-10-29 08:28:54] debug GET /favicon.ico HTTP/2.0 @ 127.0.0.1:60960
[2017-10-29 08:29:00] debug close 127.0.0.1:60960 #145
```

The config file:

```php
$host = (new Aerys\Host())
    ->expose(env('HOST_IP'), env('PORT'))
    ->name(env('HOST_NAME'))
    ->encrypt(env('SSL_CERTIFICATE'), env('SSL_KEY'))
    ->use(Aerys\root(__DIR__.'/../public'));
```

I'm getting the error when visiting `https://playground.local:4433/` with the setup found in the repo.

**Comment by kelunik** [comment]
I'm pretty sure that `v0.7.0` fucked up the port matching for named hosts without a port. @bwoebi: Could you have a look?

**Comment by bwoebi** [comment]
Fixed for now, will push a test later.

**Comment by bwoebi** [comment]
And tagged 0.7.2.

@kelunik Nah, not really related. Port reading on http/2 was always 0 here.

**Comment by brendt** [comment]
Can confirm this is fixed in 0.7.2. Thanks for the fast response guys!
